### PR TITLE
Add standard name vars

### PR DESCRIPTION
**Pull request by benjwadams**
Adds standard name variables to ERDDAP templates

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -22,3 +22,4 @@ rq-dashboard==0.3.4
 netCDF4==1.2.4
 cf-units==1.1.1
 flask-cors==3.0.2
+compliance-checker>=1.1.1
```
[the remark this answers is not in the document]
Comment (benjwadams) [reply requirements.txt]: Fixed

File: scripts/build_erddap_catalog.py [hunk 2]
```diff
@@ -155,12 +185,33 @@ def build_erddap_catalog_fragment(data_root, user, deployment, template_dir,
...
+            logger.exception("Exception occurred while generating dataset XML:")
             return templ
 
+def add_standard_name_var(var):
+    """
+    Adds an unhandled standard name variable to the ERDDAP datasets.xml
+    """
+    dvar_elem = etree.Element('dataVariable')
+    source_name = etree.SubElement(dvar_elem, 'sourceName')
+    source_name.text = var.name
```
Comment (Bobfrat) [inline scripts/build_erddap_catalog.py]: Copying over the variable name works but we may want to revisit this to try to have consistency in the variable names.

File: scripts/build_erddap_catalog.py [hunk 3]
```diff
@@ -155,12 +185,33 @@ def build_erddap_catalog_fragment(data_root, user, deployment, template_dir,
...
+
+            for var in standard_name_vars:
+                var_elem = add_standard_name_var(var)
+                tree.append(var_elem)
+
             return etree.tostring(tree)
 
         except:
-            logger.exception()
+            import ipdb; ipdb.set_trace()
```
Comment (Bobfrat) [inline scripts/build_erddap_catalog.py]: @benjwadams we'll need to remove this

Comment (benjwadams) [reply scripts/build_erddap_catalog.py]: Oops

File: scripts/build_erddap_catalog.py (hunk 2 above)
Comment (benjwadams) [reply scripts/build_erddap_catalog.py]: What are you proposing here?

Comment (Bobfrat) [reply scripts/build_erddap_catalog.py]: I was going to propose using the standard name but after discussing with @kerfoot we should just leave it as you have here for now.

File: scripts/build_erddap_catalog.py [hunk 4]
```diff
@@ -137,6 +147,26 @@ def build_erddap_catalog_fragment(data_root, user, deployment, template_dir,
...
+                         'pressure', 'temperature', 'conductivity', 'salinity',
+                         'density', 'lat', 'lon', 'time_uv', 'lat_uv',
+                         'lon_uv', 'u', 'v', 'platform', 'instrument_ctd'}
+
+    # need to explicitly cast keys to set in Python 2
+    exclude_vars = (existing_varnames | set(dest_var_remaps.keys()) |
+                    required_qc_vars | {'latitude', 'longitude'})
+
+    # need to pass None to constructor?  Why?
+    standard_name_table = util.StandardNameTable(None)
```
Comment (Bobfrat) [inline scripts/build_erddap_catalog.py]: We need to think about how we can make sure we're always using the latest standard name table

Comment (benjwadams) [reply scripts/build_erddap_catalog.py]: It should already be doing that: https://github.com/ioos/compliance-checker/blob/master/compliance_checker/cf/util.py#L263-L271

Comment (Bobfrat) [reply scripts/build_erddap_catalog.py]: That doesn't guarantee the latest standard name table. There is however a way to download a specific version but how would we know what the latest version is? 

https://github.com/ioos/compliance-checker/blob/master/cchecker.py#L107-L108

Need to scrape the CF conventions page?
http://cfconventions.org/standard-names.html

**Comment by benjwadams** [comment]
@Bobfrat, I rebased this branch to clean up some of the commit history.  Please pull down with `--force` if you want to inspect changes.

**Pull request merged**